typescript/mass-replay.md

```diff
@@ -0,0 +1,89 @@
+# Mass Replay Verification API
+
+Users often manually verify that they have not broken compat with existing
+workflow histories when making changes by downloading histories and replaying
+them with a replay worker.
+
+We aim to make that simpler and faster with the below changes, which should
+be similarly applicable to all SDKs but are laid out here in TS first.
+
+For flexibility and compose-ability reasons, it is desirable to separate the
+construction of a set of histories the user might care about from the 
+downloading of those histories and the replay of those histories. Hence
+the base level APIs proposed are (in order):
+
+## Fetching
+See https://github.com/temporalio/proposals/pull/67
+
+Which includes functions for listing workflows according to some query params.
+Those functions return something like `Iterable<WorkflowExecutionInformation>`
+
+## Downloading
+This is simply a matter of (lazily) mapping the iterable:
+`Iterable<WorkflowExecutionInformation> -> AsyncIterable<WorkflowIDAndHistory>`
+
+Preserving the workflow id is important, since history does not include it
+by itself.
+
+By itself, this would look something like:
+`const historyIterator = client.workflow.downloadLazily(workflowIds)`
+
+Which would be sugar for creating a returning an async iterator which calls
+`getExecutionHistory` on each id. This might include doing so concurrently
```

Comment:
Pretty easy to limit the concurrency with rxjs, it's built into `flatMap`.

Reply:
It's not just a matter of that, it's also a matter of buffering that and sending it down the pipe into the rust side appropriately. Shouldn't be all that hard though.